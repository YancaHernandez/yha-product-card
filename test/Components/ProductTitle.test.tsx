import React from 'react';
import renderer from 'react-test-renderer';
import { ProductTitle, ProductCard } from '../../src/components';

import { product1 } from "../data/products";
describe('ProductTitle', () => {
    it('debe de mostrar el componente con el titulo personalizado', () => {
        const wrapper = renderer.create(<ProductTitle title="Titulo personalizado" />);

        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    test('debe de mostrar el componente con el titulo por defecto', () => {
        const wrapper = renderer.create(
            <ProductCard product={product1} >
                {
                    () => <ProductTitle />
                }
            </ProductCard>
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});