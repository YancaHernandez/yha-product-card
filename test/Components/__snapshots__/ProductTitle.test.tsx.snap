// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle debe de mostrar el componente con el titulo personalizado 1`] = `
<span
  className="productDescription undefined"
>
  Titulo personalizado
</span>
`;

exports[`ProductTitle debe de mostrar el componente con el titulo por defecto 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - Sin Imagen
  </span>
</div>
`;
